src/index/vchordrq/am/am_build.rs

```diff
@@ -1324,6 +1372,201 @@ fn make_default_build(
     }]
 }
 
+#[derive(Debug, PartialEq)]
+struct PriorityItem {
+    index: usize,
+    priority: f64,
+}
+
+impl Eq for PriorityItem {}
+
+impl PartialOrd for PriorityItem {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for PriorityItem {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        if self.priority.is_nan() {
+            std::cmp::Ordering::Less
+        } else if other.priority.is_nan() {
+            std::cmp::Ordering::Greater
+        } else {
+            match self.priority.partial_cmp(&other.priority) {
+                Some(ordering) => ordering,
+                None => std::cmp::Ordering::Equal,
+            }
+        }
+    }
+}
+
+fn make_internal_approximate_build(
```

Comment:
We might prefer to move the KMeans-related code to `crates/k_means`.

Reply:
This is the designed changes:
- `k_means_lookup` is moved into `crates/k_means`
- `k_means_centroids_inplace` is created in `crates/k_means`, used to calculate centroids from sum/count
- `sainte_lague_allocate` is moved to `crates/k_means`, based on https://en.wikipedia.org/wiki/Sainte-Lagu%C3%AB_method